packages/jest-runtime/src/shouldLoadAsEsm.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {dirname, extname} from 'path';
+// @ts-ignore: experimental, not added to the types
+import {SourceTextModule} from 'vm';
+import type {Config} from '@jest/types';
+import readPkgUp = require('read-pkg-up');
+
+export const runtimeSupportsVmModules = typeof SourceTextModule === 'function';
+
+// this is a bad version of what https://github.com/nodejs/modules/issues/393 would provide
+export default function shouldLoadAsEsm(path: Config.Path): boolean {
```

Comment:
My main concern with this PR is this, and if it affects performance. It'll be called for every setupfile per test file. I'm not sure how to cache this in a way that's safe in watch mode

Reply:
solved this by moving it to `jest-resolve` which already has a cache busting mechanism